msdk-rawdata/msdk-rawdata-filters/src/test/java/io/github/msdk/filtering/scanfilters/MeanFilterMethodTest.java

```diff
@@ -0,0 +1,110 @@
+import io.github.msdk.datamodel.impl.MSDKObjectBuilder;
+import io.github.msdk.datamodel.msspectra.MsIon;
+import io.github.msdk.datamodel.msspectra.MsSpectrumDataPointList;
+import io.github.msdk.datamodel.rawdata.MsScan;
+import io.github.msdk.datamodel.rawdata.RawDataFile;
+import io.github.msdk.io.mzml.MzMLFileImportMethod;
+import java.io.File;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class MeanFilterMethodTest {
+
+    private static final String TEST_DATA_PATH = "src/test/resources/";
+
+    @Test
+    public void testMeanFilter() throws MSDKException {
+       
+        // Import the file
+        File inputFile = new File(TEST_DATA_PATH + "orbitrap_300-600mz.mzML");
+        Assert.assertTrue("Cannot read test data", inputFile.canRead());
+        MzMLFileImportMethod importer = new MzMLFileImportMethod(inputFile);
+        RawDataFile rawFile = importer.execute();
+        Assert.assertNotNull(rawFile);
+        Assert.assertEquals(1.0, importer.getFinishedPercentage(), 0.0001);
+
+        // Create the data needed by the Mean Filter Method
+        List<MsScan> scans = rawFile.getScans();
+        DataPointStore store = DataPointStoreFactory.getMemoryDataStore();
+
+        // Testing the filter with normal values
+        for (MsScan scan : scans) {
+            MeanFilterMethod meanFilter = new MeanFilterMethod(scan, 0.5, store);
+            MsScan newScan = meanFilter.execute();
+            Assert.assertEquals(1.0, meanFilter.getFinishedPercentage(), 0.0001);
+
+            // The result of the method can't be Null
+            Assert.assertNotNull(newScan);
+        }
+
+        // Test windowLength == 0 -> the resulting scan should be the equal to the input scan
+        MsScan scanTest = scans.get(0);
+        MeanFilterMethod meanFilter = new MeanFilterMethod(scanTest, 0, store);
+        MsScan newScan = meanFilter.execute();
+        Assert.assertEquals(1.0, meanFilter.getFinishedPercentage(), 0.0001);
+
+        // The result of the method can't be Null
+        Assert.assertNotNull(newScan);
+
+        // The resulting scan should be equal to the input scan
+        MsSpectrumDataPointList dataPoints = MSDKObjectBuilder.getMsSpectrumDataPointList();
+        scanTest.getDataPoints(dataPoints);
+       
+        float intensityAverage = 0;
+
+        for (MsIon ion : dataPoints) {
+            intensityAverage += ion.getIntensity();
+        }
+        
+        intensityAverage/=dataPoints.getSize();
+        
+        // Get the mz and intensities values from the input scan
+        double mzValues[] = dataPoints.getMzBuffer();
+        float intensityValues[] = dataPoints.getIntensityBuffer();
+        int numberOfDataPoints = dataPoints.getSize();
+        
+        newScan.getDataPoints(dataPoints);
+        
+        // Get the mz and intensities values from the filtered scan
+        double newMzValues[] = dataPoints.getMzBuffer();
```

Comment:
This call to dataPoints.getMzBuffer() is going to return the same double[] array as the previous call a few lines above - so your comparison will always be true. You will need to create two separate MsSpectrumDataPointList instances, one for the original data points and one for the filtered ones.